### PR TITLE
Last Leg of Porting Stuffs

### PR DESCRIPTION
**Pull request by meawoppl**
This is a file-complete port with a bunch of shortcomings, posted in current form for discussion only.

### CODE REVIEW

File: src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+    Triggerable configurePatternAsTriggerable(EntwinedTriggerablePattern pattern) {
+        LXChannel channel = lx.engine.mixer.addChannel(new EntwinedBasePattern[] {pattern});
+        setupChannel(channel, false);
+
+        pattern.onTriggerableModeEnabled();
+        return pattern.getTriggerable();
+    }
+
+    void setupChannel(final LXChannel channel, boolean noOpWhenNotRunning) {
+        // @Slee, honestly no idea what the intention is here...
```
Comment (mcslee) [inline src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java:272]: If you want to add a custom transition type to the list of available types in the dropdown, you can do that in your initialize() callback via:

lx.registry.addChannelBlend()
http://lx.studio/api/heronarts/lx/LXRegistry.html#addChannelBlend-java.lang.Class-

File: src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+
+    void registerVisual(
+            EntwinedTriggerablePattern pattern,
+            String nfcSerialNumber,
+            int apc40DrumpadRow,
+            VisualType visualType) {
+
+        LXBlend t = new DissolveBlend(lx);
+
+        // NOTE(meawoppl) @Slee same question below.  re `.setDuration(dissolveTime);`
```
Comment (mcslee) [inline src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java:245]: There are no longer per-pattern transition settings. The transition type and duration are now simply set at the channel level.

File: src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+
+        channel.addListener(
+                new LXChannel.AbstractListener() {
+                    LXTransition transition;
+
+                    @Override
+                    public void patternWillChange(
+                            LXChannel channel, LXPattern pattern, LXPattern nextPattern) {
+                        if (!channel.enabled.isOn()) {
+                            transition = nextPattern.getTransition();
```
Comment (mcslee) [inline src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java:290]: Per the above comment about per-pattern transitions going away, this is maybe just all not relevant anymore. I'd just pick a sensible transition to assign at the channel level and leave it.

File: src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                    (parameter) -> channel.enabled.setValue(channel.fader.getValue() != 0));
+        }
+    }
+
+    void registerPatternController(String name, EntwinedBasePattern pattern) {
+        // @Slee is there a modern version of `.setDuration(dissolveTime);` on
+        // blends?
+        LXBlend t = new DissolveBlend(lx);
+
+        // @Slee not sure where `.setTranstion` got to...
```
Comment (mcslee) [inline src/main/java/com/charlesgadeken/entwined/EntwinedTriggers.java:316]: Yep it's gone, only exists at channel level.

**Comment by meawoppl** [comment]
Subsumed by #24

**Pull request closed without merging**